internal/services/maintenance/public_maintenance_configurations_data_source.go

```diff
@@ -0,0 +1,215 @@
+package maintenance
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/maintenance/mgmt/2021-05-01/maintenance"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+const recurMondayToThursday = "Monday-Thursday"
+const recurFridayToSunday = "Friday-Sunday"
+
+func dataSourcePublicMaintenanceConfigurations() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Read: dataSourcePublicMaintenanceConfigurationsRead,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+
+			"location": {
+				Type:      pluginsdk.TypeString,
+				Optional:  true,
+				StateFunc: azure.NormalizeLocation,
+			},
+
+			"scope_filter": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"All", // All is still accepted by the API
+					string(maintenance.ScopeExtension),
+					string(maintenance.ScopeHost),
+					string(maintenance.ScopeInGuestPatch),
+					string(maintenance.ScopeOSImage),
+					string(maintenance.ScopeSQLDB),
+					string(maintenance.ScopeSQLManagedInstance),
+				}, false),
+			},
+
+			"recur_every_filter": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					recurMondayToThursday,
+					recurFridayToSunday,
+				}, false),
+			},
+
+			"public_maintenance_configurations": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"location": {
```

Comment:
we should have white space here?
```suggestion
						},

						"id": {
							Type:     pluginsdk.TypeString,
							Computed: true,
						},

						"location": {
```